### ISSUE TITLE
Clarify differences between nomad_volume and nomad_external_volume

### ISSUE THREAD

**Issue by apollo13** [hashicorp/terraform-provider-nomad#252]
Looking at the docs it is not really clear what the differences between `nomad_volume` and `nomad_external_volume` are. From what I gathered looking at the arguments `nomad_volume` requires `external_id`, so it seems to map to csi `register` whereas `nomad_external_volume` seems to use `create` to create a volume. All in all it came as a surprise that `nomad_external_volume` is actually the "more internal" variant of those two :)

**Comment by lgfa29** [comment]
> From what I gathered looking at the arguments `nomad_volume` requires `external_id`, so it seems to map to csi `register` whereas `nomad_external_volume` seems to use `create` to create a volume.

That's pretty much it. Any suggestion on how the description could be improved?

> All in all it came as a surprise that `nomad_external_volume` is actually the "more internal" variant of those two :)

Hum...why is it more internal? `nomad_volume` creates a volume in Nomad, while `nomad_external_volume` creates a volume in an external storage provider then registers it in Nomad.

> If you create a volume using `nomad_volume`, then deregister it (eg outside of terraform) and `terraform apply` again, its not possible to reuse the same external volume

You shouldn't really be changing things outside Terraform 😅  but this should work. Could you provide a reproduction?

> whereas with `nomad_external_volume`, its not possible to create a volume.

`nomad_external_volume` does create volumes using the CSI plugin deployed in your cluster. Could you provide more details?

**Comment by apollo13** [comment]
> That's pretty much it. Any suggestion on how the description could be improved?

Good question, did those get updated recently? I could have sworn that they said the same thing when I opened this issue -- but maybe I mixed up the tabs. Sorry for the noise.

> Hum...why is it more internal? `nomad_volume` creates a volume in Nomad, while `nomad_external_volume` creates a volume in an external storage provider then registers it in Nomad.

I guess when you say "creates a volume in Nomad" you mean register but I think then of "creates via CSI". Either way, since both are CSI volumes for me it would have made more sense if `nomad_volume` created & registered the volume and `nomad_external_volume` only registered it. The name `external` here implies to me that it is external (ie already existing). As for the "more internal" part I should have probably said "more integrated". Nomad only got support for volume creation recently but it feels like that is the recommended way (especially since it is often not that easy to create volumes manually via a CSI plugin).

Either way, all good -- I'll close this one out since I probably was looking at the same docs page twice when opening the issue.
